nomos-services/key-management-system/src/keys/ed25519.rs

```diff
@@ -1,20 +1,49 @@
-use bytes::Bytes;
+use bytes::Bytes as RawBytes;
 use ed25519_dalek::ed25519::signature::Signer as _;
-use overwatch::DynError;
 use serde::{Deserialize, Serialize};
 use zeroize::ZeroizeOnDrop;
 
-use crate::secure_key::SecuredKey;
+use crate::{
+    encodings::Encoding,
+    keys::{secured_key::SecuredKey, KeyError},
+};
+
+type InnerEncoding = crate::encodings::Bytes;
 
 #[derive(Serialize, Deserialize, ZeroizeOnDrop)]
 pub struct Ed25519Key(pub(crate) ed25519_dalek::SigningKey);
 
-impl SecuredKey for Ed25519Key {
-    fn sign(&self, data: Bytes) -> Result<Bytes, DynError> {
-        Ok(Bytes::copy_from_slice(&self.0.sign(&data).to_bytes()))
+impl SecuredKey<InnerEncoding> for Ed25519Key {
+    type Signature = InnerEncoding;
+    type PublicKey = InnerEncoding;
+    type Error = KeyError;
+
+    fn sign(&self, data: &InnerEncoding) -> Result<Self::Signature, Self::Error> {
+        let data_bytes = data.as_bytes();
+        let signature_slice = self.0.sign(data_bytes).to_bytes();
+        let signature_bytes = RawBytes::copy_from_slice(&signature_slice);
+        Ok(Self::Signature::from(signature_bytes))
+    }
+
+    fn as_public_key(&self) -> Self::PublicKey {
+        let verifying_key = self.0.verifying_key();
+        let verifying_key_bytes = RawBytes::copy_from_slice(verifying_key.as_bytes());
+        Self::PublicKey::from(verifying_key_bytes)
+    }
+}
+
+impl SecuredKey<Encoding> for Ed25519Key {
+    type Signature = Encoding;
+    type PublicKey = Encoding;
+    type Error = KeyError;
+
+    fn sign(&self, data: &Encoding) -> Result<Self::Signature, Self::Error> {
+        match data {
+            Encoding::Bytes(data) => self.sign(data).map(Encoding::from),
+        }
     }
```

Comment:
Is there 2 impl blocks for Ed25519Key? This second one is for the wrapper enum right?